docs/controllers/availability.md

```diff
@@ -0,0 +1,92 @@
+# Availability
+
+This chapter is about strategies for improving controller availability and tail latencies.
+
+## Motivation
+
+Despite the common goals often set forth for application deployments, most `kube` controllers:
+
+- can run in a single replica (default recommendation)
+- can handle being killed, and be shifted to another node
+- can handle minor downtime
+
+This is due to a couple of properties:
+
+- Rust images are often very small and will reschedule quickly
+- watch streams re-initialise quickly with the current state on boot
+- [[reconciler#idempotency]] means multiple repeat reconciliations are not problematic
+- parallel execution mode of reconciliations makes restarts fast
+
+These properties combined creates a low-overhead system that is normally quick to catch-up after being rescheduled, and offers a traditional Kubernetes __eventual consistency__ guarantee.
+
+That said, this setup can struggle under strong consistency requirements. Ask yourself:
+
+- How fast do you expect your reconciler to react?
+- Can you allow `30s` P95 downtimes from reschedules?
+
+## Reactivity
+
+If you want to improve __average reactivity__, then traditional [[scaling]] and [[optimization]] strategies can help:
+
+- Configure controller concurrency to avoid waiting for a reconciler slot
+- Optimize the reconciler, avoid duplicated work
+- Satisfy CPU requirements to avoid cgroup throttling
+
+You can plot heatmaps of reconciliation times in grafana using standard [[observability#What Metrics]].
+
+<!--TODO: can we measure time from watch event seen to watch event received by reconciler?-->
+
+## High Availability
+
+At a certain point, the slowdown caused by pod reschedules is going to dominate the latency metrics. Thus, having more than one replica (and having HA) is a requirement for further reducing tail latencies.
```

Comment:
TIOLI: Is it worth pointing or making explicit that HA also avoids having a single point of failure? I'll leave the judgement up to you, just driving by.

Reply:
yeah, i had totally neglected the redundancy aspect. have restructured the HA section now more focusing on why HA is different than normal apps and how. have not explicitly mentioned SPOF because it feels a bit loaded (all controllers are in a sense SPOFS of their own domain and a replica is only addressing the speed of that).

I did find a way to do a shoutout for the rollout problem though. Anyway https://github.com/kube-rs/website/pull/58/commits/7cca11dca720ee322c7f45a1a74d7fa953d01e8b